[Bug]: `test_snmp_queue_counters.py` fails with `Failed: Buffer Queue list can't be empty if valid interface is selected.`

```
        duthost.shell(cmd)
        data = json.loads(duthost.shell("cat {}".format(ORIG_CFG_DB),
                                        verbose=False)['stdout'])
        buffer_queue_to_del = None
        # Get appropriate buffer queue value to delete
        buffer_queues = list(data['BUFFER_QUEUE'].keys())
        iface_buffer_queues = [bq for bq in buffer_queues if bq.split('|')[0] == interface]
        if iface_buffer_queues:
            buffer_queue_to_del = iface_buffer_queues[0]
        else:
>           pytest_assert(False, "Buffer Queue list can't be empty if valid interface is selected.")
E           Failed: Buffer Queue list can't be empty if valid interface is selected.
```

This failure was introduced by https://github.com/sonic-net/sonic-mgmt/pull/16687 
This PR doesn't account for chassis systems
From the description:
`buffer_queues = ['Ethernet112|0-1', 'Ethernet112|2-4', 'Ethernet112|5', 'Ethernet112|6', 'Ethernet112|7', 'Ethernet116|0-1', ...]`

From the failure:
`buffer_queues = ['nfc405-3|Asic0|Ethernet0|0-2', 'nfc405-3|Asic0|Ethernet0|3-4', 'nfc405-3|Asic0|Ethernet0|5-6', 'nfc405-3|Asic0|Ethernet100|0-2', 'nfc405-3|Asic0|Ethernet100|3-4', 'nfc405-3|Asic0|Ethernet100|5-6', ...]`

On chassis the buffer_queue includes the full system port name (LC + ASIC).
So this indexing of the first set of characters before the `|` character won't work for chassis:
`iface_buffer_queues = [bq for bq in buffer_queues if bq.split('|')[0] == interface]`

--- a/tests/snmp/test_snmp_queue_counters.py
+++ b/tests/snmp/test_snmp_queue_counters.py
@@ -126,7 +126,7 @@ def test_snmp_queue_counters(duthosts,
 
     # Get appropriate buffer queue value to delete
     buffer_queues = list(data['BUFFER_QUEUE'].keys())
-    iface_buffer_queues = [bq for bq in buffer_queues if bq.split('|')[0] == interface]
+    iface_buffer_queues = [bq for bq in buffer_queues if any(val == interface for val in bq.split('|'))]
     if iface_buffer_queues:
         buffer_queue_to_del = iface_buffer_queues[0]
     else:

